src/rest-client.ts

```diff
@@ -1887,68 +1719,28 @@ export class RestClient extends BaseRestClient {
     return this.get('/api/v5/market/books-full', params);
   }
 
-  /**
-   * @deprecated Breaking change: this function will change soon to accept a single object for all REST API parameters.
-   * See issue https://github.com/tiagosiebler/okx-api/issues/88 for more details.
-   */
-  getCandles(
-    instId: string,
-    bar: string = '1m',
-    pagination?: Pagination,
-  ): Promise<Candle[]> {
-    return this.get('/api/v5/market/candles', {
-      instId,
-      bar,
-      ...pagination,
-    });
-  }
-
-  getCandlesV2(params: CandleRequest): Promise<Candle[]> {
+  getCandles(params: CandleRequest): Promise<Candle[]> {
     return this.get('/api/v5/market/candles', params);
   }
 
-  /**
-   * @deprecated Breaking change: this function will change soon to accept a single object for all REST API parameters.
-   * See issue https://github.com/tiagosiebler/okx-api/issues/88 for more details.
-   */
-  getHistoricCandles(
-    instId: string,
-    bar: string = '1m',
-    pagination?: Pagination,
-  ): Promise<Candle[]> {
-    return this.get('/api/v5/market/history-candles', {
-      instId,
-      bar,
-      ...pagination,
-    });
-  }
-
-  getHistoricCandlesV2(params: CandleRequest): Promise<Candle[]> {
+  getHistoricCandles(params: CandleRequest): Promise<Candle[]> {
     return this.get('/api/v5/market/history-candles', params);
   }
 
-  /**
-   * @deprecated Breaking change: this function will change soon to accept a single object for all REST API parameters.
-   * See issue https://github.com/tiagosiebler/okx-api/issues/88 for more details.
-   */
-  getTrades(instId: string, limit?: number): Promise<Trade[]> {
-    return this.get('/api/v5/market/trades', { instId, limit });
+  getTrades(params: { instId: string; limit?: number }): Promise<Trade[]> {
+    return this.get('/api/v5/market/trades', params);
   }
 
-  /**
-   * @deprecated Breaking change: this function will change soon to accept a single object for all REST API parameters.
-   * See issue https://github.com/tiagosiebler/okx-api/issues/88 for more details.
-   */
-  getHistoricTrades(
-    instId: string,
+  getHistoricTrades(params: {
+    instId: string;
     pagination?: {
       after?: numberInString;
       before?: numberInString;
       limit?: numberInString;
       type?: '1' | '2';
-    },
-  ): Promise<Trade[]> {
-    return this.get('/api/v5/market/history-trades', { instId, ...pagination });
+    };
```

Comment:
^ I _think_ (please double check), this pagination object needs to be merged into params.

```
getHistoricTrades(params: {
    instId: string;
    after?: numberInString;
    before?: numberInString;
    limit?: numberInString;
    type?: '1' | '2';
}): Promise....
```